File: Spring/haru/src/main/java/com/ssafy/haru/service/HomeworkService.java
package com.ssafy.haru.service;

import com.ssafy.haru.model.dto.Homework;

public interface HomeworkService {
    // 숙제 조회
    Homework select(String userId, String homeworkDate);

    // 숙제 저장
    boolean insert(Homework homework);

    // 숙제 수정
    boolean update(Homework homework);

    // 숙제 삭제
    boolean delete(String userId, String homeworkDate);
}

File: android/app/src/main/java/com/ssafy/diary/api/HomeworkService.kt
package com.ssafy.diary.api

import com.ssafy.diary.dto.Homework
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Query

interface HomeworkService {
    // 숙제 조회
    @GET("api/homework/select")
    suspend fun getHomework(@Query("userId") userId: String, @Query("homeworkDate") homeworkDate: String): Response<Homework>

    // 숙제 저장
    @POST("api/homework/insert")
    suspend fun saveHomework(@Body homework: Homework): Response<Boolean>

    // 숙제 수정
    @PUT("api/homework/update")
    suspend fun updateHomework(@Body homework: Homework): Response<Boolean>

    // 숙제 삭제
    @DELETE("api/homework/delete")
    suspend fun deleteHomework(@Query("userId") userId: String, @Query("homeworkDate") homeworkDate: String): Response<Boolean>
}